tools/ci_build/build.py

```diff
@@ -1183,19 +1184,6 @@ def generate_build_tree(
 
     if is_macOS() and not args.android:
         cmake_args += ["-DCMAKE_OSX_ARCHITECTURES=" + args.osx_arch]
-        if args.use_xcode:
```

Comment:
this check seems irrelevant now since our minimum cmake version higher than the ones handled here.